src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/InProcDataCollectionExtensionManager.cs

```diff
@@ -46,62 +60,31 @@ public InProcDataCollectionExtensionManager(string runSettings, ITestRunCache te
 
             // Initialize InProcDataCollectors
             this.InitializeInProcDataCollectors(runSettings);
-        }
-
-        public bool IsInProcDataCollectionEnabled { get; private set; }
-
-        /// <summary>
-        /// Loads all the inproc data collector dlls
-        /// </summary>       
-        private void InitializeInProcDataCollectors(string runSettings)
-        {
-            try
-            {
-                // Check if runsettings contains in-proc datacollector element
-                var inProcDataCollectionRunSettings = XmlRunSettingsUtilities.GetInProcDataCollectionRunSettings(runSettings);
-                var inProcDataCollectionSettingsPresentInRunSettings = inProcDataCollectionRunSettings?.IsCollectionEnabled ?? false;
 
-                // Verify if it has any valid in-proc datacollectors or just a dummy element
-                inProcDataCollectionSettingsPresentInRunSettings = inProcDataCollectionSettingsPresentInRunSettings &&
-                    inProcDataCollectionRunSettings.DataCollectorSettingsList.Any();
+            FlushResultTestResultPoperty = TestProperty.Register(id: "allowTestResultFlush", label: "allowTestResultFlush", category: string.Empty, description: string.Empty, valueType: typeof(bool), validateValueCallback: null, attributes: TestPropertyAttributes.None, owner: typeof(TestCase));
 
-                // Initialize if we have atleast one
-                if (inProcDataCollectionSettingsPresentInRunSettings)
-                {
-                    this.inProcDataCollectorSettingsCollection = inProcDataCollectionRunSettings.DataCollectorSettingsList;
-
-                    var interfaceTypeInfo = typeof(InProcDataCollection).GetTypeInfo();
-                    foreach (var inProcDc in this.inProcDataCollectorSettingsCollection)
-                    {
-                        var inProcDataCollector = this.CreateDataCollector(inProcDc, interfaceTypeInfo);
-                        this.inProcDataCollectors[inProcDataCollector.AssemblyQualifiedName] = inProcDataCollector;
-                    }
-                }
-            }
-            catch (Exception ex)
-            {
-                EqtTrace.Error("InProcDataCollectionExtensionManager: Error occured while Initializing the datacollectors : {0}", ex);
-            }
-            finally
-            {
-                this.IsInProcDataCollectionEnabled = this.inProcDataCollectors.Any();
-            }
+            dataCollectionTestCaseEventManager.TestCaseEnd += this.TriggerTestCaseEnd;
```

Comment:
Are the events subscribed even when Initialize fails? What if there are no inproc data collectors enabled?